kolibri/core/content/hooks.py

```diff
@@ -108,3 +109,17 @@ def render_to_page_load_async_html(self):
             )
         ]
         return mark_safe("\n".join(tags))
+
+
+class ContentNodeDisplayHook(KolibriHook):
+    """
+    A hook that registers a capability of a plugin to provide a user interface
+    for a content node. When subclassed, this hook should expose a method that
+    accepts a ContentNode instance as an argument, and returns a URL where the
```

Comment:
Originally, I had read this comment and interpreted it to mean that the hooks expose a URL like

```/viewcontent/[node_id]```

I didn’t see that it was in fact more like

```/viewcontent?node_id=[node_id]&channel_id=[channel_id]&content_id=[content_id]```

This was the source of some of my confusion in my previous review comments, and makes me feel better about this PR, so I'll approve it.

That said, I do think there is an argument to have the plugins responsible for turning the GET params directly into a local URL, rather than having the `ContentNodeDisplayHook` perform resolution through an intermediary channel node object. Channel nodes are not always be the appropriate means of content identification for resolving the final plugin URL (e.g. in a lesson context, quiz, local curation, etc).

Another possible solution would be to have some fixed vocabulary of GET params defined in core, e.g. `channel_node_id`, `content_id`, `lesson_id`, etc. Then, each plugin would have direct access to the params, and can do its own DB lookups rather than relying on being passed a `ContentNode` instance. Then, plugins could implement their own handling for cases like `(content_id, lesson_id)` versus `(channel_node_id,)`. This would allow for similar resolution behavior that @jamalex [described below](https://github.com/learningequality/kolibri/pull/5641/files#r292227054).

Reply:
Yeah - we can definitely iterate on the hook implementation, as long as we are fine with the URL scheme, because that's the thing we shouldn't change!